src/hooks/usePopup.ts

```diff
@@ -26,15 +26,16 @@ export const usePopup = <P, K extends keyof P>(
 
     const open = useCallback<OptionalParamFunction<Omit<P, K>, void>>(
         (omittedProps?: Omit<P, K>) => {
-            if (!getPopup(popupIdentifier.current)) {
-                const popup = new DefaultPopup(
-                    PopupComponent,
-                    { ...props, ...omittedProps } as P,
+            const popup =

```

Comment:
It is better to always create a new popup passing props in `DefaultPopup`. In that case, `setProps` function is unnecessary.